### PR TITLE
main/lua-xml: Several changes

### PR DESCRIPTION
**Pull request by leitao**
Several changes to the lua-xml package, that now is named luaxml.
These are the changes:
  * Upgrade to the latest version
  * Change project URL (now on github)
  * Rebased the patch
  * Changed $_builddir to $builddir

### CODE REVIEW

File: main/lua-xml/APKBUILD [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Maintainer: Natanael Copa <ncopa@alpinelinux.org>
 _luaversions="5.1 5.2"
-pkgname=lua-xml
+pkgname=luaxml
```
Comment (jirutka) [inline main/lua-xml/APKBUILD]: All Lua libraries are named as `lua-<name>` or lua<lver>-<name>`, where `<lver>` is Lua version (e.g. lua-xml, lua5.1-xml, …).

File: main/lua-xml/APKBUILD [hunk 2]
```diff
@@ -17,28 +17,28 @@ for _v in $_luaversions; do
...
 done
 
-source="http://viremo.eludi.net/LuaXML/LuaXML_$pkgver.zip
-0001-Fix-error-where-TAG-should-be-xml.TAG.patch"
-
-_builddir="$srcdir"
+source="$pkgname-$pkgver.tar.gz::https://github.com/LuaDist/luaxml/archive/$pkgver.tar.gz
+	0001-Fix-error-where-TAG-should-be-xml.TAG.patch"
 
+builddir="$srcdir"
```
Comment (jirutka) [inline main/lua-xml/APKBUILD]: This should point to the root of the unpacked package, so  `$srcdir/$pkgname-$pkgver`.

File: main/lua-xml/APKBUILD [hunk 3]
```diff
@@ -17,28 +17,28 @@ for _v in $_luaversions; do
...
 prepare() {
-	cd "$_builddir"
+	cd "$builddir"
 	for i in $source; do
 		case $i in
 		*.patch) msg $i; patch -p1 -i "$srcdir"/$i || return 1;;
 		esac
 	done
```
Comment (jirutka) [inline main/lua-xml/APKBUILD]: This is unnecessary. Applying of patches is already handled by the `default_prepare` function that is called when no custom `prepare` function is declared. When custom function is declared, just call `default_prepare`, usually as the first command of the function.

File: main/lua-xml/APKBUILD [hunk 4]
```diff
@@ -17,28 +17,28 @@ for _v in $_luaversions; do
...
 
+	dest="$pkgname-$pkgver"
```
Comment (jirutka) [inline main/lua-xml/APKBUILD]: Custom variables inside functions should be always declared as `local`, if not needed in more functions. Moreover, this is totally unnecessary, just use `buildddir` properly.

**Comment by leitao** [comment]
hi @jirutka. Thanks for your review. I just updated the commit to adjust to your recommendations.

**Comment by leitao** [comment]
hi @jirutka, any feedback about this new version?

**Comment by jirutka** [comment]
Sorry, I forgot to this patch, I'll look at it tomorrow.

**Comment by jirutka** [comment]
The current state of LuaXML is really a mess, there are few forks and it’s not clear what is actually the original LuaXML, since the upstream source is down. However, it seems that https://github.com/LuaDist2/luaxml corresponds to the version we currently have and also what is on LuaRocks. There are many changes between this and https://github.com/LuaDist/luaxml, even different versioning scheme.

So I prefer to use https://github.com/LuaDist2/luaxml as new source. I’ve already updated lua-xml in 5a461824f433db3d66099e34f371c4608a93fed0, 6a1e7a93bbc27a044874af176b4cd32f42c69e08 and 6a672cdda3dc2c560540ccf164b1d18ea8105906.

**Pull request closed without merging**